src/openrct2/peep/Peep.cpp

```diff
@@ -989,13 +989,20 @@ static void sub_68F41A(rct_peep * peep, sint32 index)
 
         if (peep->peep_flags & PEEP_FLAGS_EXPLODE && peep->x != LOCATION_NULL)
         {
-            audio_play_sound_at_location(SOUND_CRASH, peep->x, peep->y, peep->z);
+            if (peep->state == PEEP_STATE_WALKING || peep->state == PEEP_STATE_SITTING)
+            {
+                audio_play_sound_at_location(SOUND_CRASH, peep->x, peep->y, peep->z);
 
-            sprite_misc_explosion_cloud_create(peep->x, peep->y, peep->z + 16);
-            sprite_misc_explosion_flare_create(peep->x, peep->y, peep->z + 16);
+                sprite_misc_explosion_cloud_create(peep->x, peep->y, peep->z + 16);
+                sprite_misc_explosion_flare_create(peep->x, peep->y, peep->z + 16);
 
-            peep_remove(peep);
-            return;
+                peep_remove(peep);
+                return;
+            }
+            else
+            {
+                peep->peep_flags &= ~PEEP_FLAGS_EXPLODE;
```

Comment:
This might not be necessary. Leaving this out would just mean that guests would explode once they leave a ride. (Or any other instance when changing their state to walking, such as entering the park)

Reply:
Correct, and that may take quite a while. A peep could for example enter a queue and then enter a ride, and then explode 10 minutes after the cheat has been used. I think it's better if the cheat is fully executed within a short time frame.